src/codeql.ts

```diff
@@ -81,13 +81,17 @@ libraryPathDependencies: codeql-${language}`
 }
 
 async function downloadDatabase(
-  token: string,
+  signedAuthToken: string,
   repoId: number,
-  language: string
+  language: string,
+  pat?: string
 ): Promise<string> {
+  const authHeader =
+    pat === undefined ? `RemoteAuth ${signedAuthToken}` : `token ${pat}`;
```

Comment:
This may need to handle the case where both `pat` and `signedAuthToken ` are null, or add a check against that.

Reply:
And in that case no auth header?

Reply:
I think no auth header in that case would be valid. And for a public repo could actually succeed at downloading the database.